CorsixTH/Lua/humanoid_actions/walk.lua

```diff
@@ -111,6 +111,10 @@ local flag_flip_h = 1
 local navigateDoor
 
 local function action_walk_raw(humanoid, x1, y1, x2, y2, map, timer_fn)
+  -- lewri: for some reason these must always make up factor*quantity = 8 or the
```

Comment:
I think this might be me more asking why. I can trim it to state the fact only instead if needed.

Reply:
My guess is that a tile walk animation has some `F` frames, and in those frames the animated NPC must move from center of the source tile to center of the destination tile, which is some `X` pixels horizontally and some `Y` pixels vertically.

Apparently this results in the stated numbers.

Reply:
I think theoretically you don't need to track the NPC position within a tile.
As far as I can see, you should be able to derive that from eg the position of the head sprite.
I tried that a few times, but so far no luck getting a useful result.